pkg/scheduler/objects/application.go

```diff
@@ -816,144 +804,155 @@ func (sa *Application) IsReservedOnNode(nodeID string) bool {
-			zap.String("appID", sa.ApplicationID),
-			zap.Stringer("node", node),
-			zap.Stringer("ask", ask))
-		return 0, fmt.Errorf("reservation key failed node or ask are nil for appID %s", sa.ApplicationID)
+func (sa *Application) unReserveInternal(reserve *reservation) int {
+	// this should not happen
+	if reserve == nil {
+		return 0
 	}
 	// unReserve the node before removing from the app
-	var num int
-	var err error
-	if num, err = node.unReserve(sa, ask); err != nil {
-		return 0, err
-	}
+	num := reserve.node.unReserve(reserve.alloc)
 	// if the unreserve worked on the node check the app
-	if _, found := sa.reservations[resKey]; found {
+	if _, found := sa.reservations[reserve.allocKey]; found {
 		// worked on the node means either found or not but no error, log difference here
 		if num == 0 {
 			log.Log(log.SchedApplication).Info("reservation not found while removing from node, app has reservation",
 				zap.String("appID", sa.ApplicationID),
-				zap.String("nodeID", node.NodeID),
-				zap.String("ask", ask.GetAllocationKey()))
+				zap.String("nodeID", reserve.nodeID),
+				zap.String("alloc", reserve.allocKey))
 		}
-		delete(sa.reservations, resKey)
-		log.Log(log.SchedApplication).Info("reservation removed successfully", zap.String("node", node.NodeID),
-			zap.String("app", ask.GetApplicationID()), zap.String("ask", ask.GetAllocationKey()))
-		return 1, nil
+		delete(sa.reservations, reserve.allocKey)
+		log.Log(log.SchedApplication).Info("reservation removed successfully",
+			zap.String("appID", sa.ApplicationID),
+			zap.String("node", reserve.nodeID),
+			zap.String("alloc", reserve.allocKey))
+		return 1
 	}
 	// reservation was not found
 	log.Log(log.SchedApplication).Info("reservation not found while removing from app",
 		zap.String("appID", sa.ApplicationID),
-		zap.String("nodeID", node.NodeID),
-		zap.String("ask", ask.GetAllocationKey()),
+		zap.String("node", reserve.nodeID),
+		zap.String("alloc", reserve.allocKey),
 		zap.Int("nodeReservationsRemoved", num))
-	return 0, nil
+	return 0
 }
 
-// Return the allocation reservations on any node.
-// The returned array is 0 or more keys into the reservations map.
+// IsAllocationReserved returns true if the allocation has been reserved
+func (sa *Application) IsAllocationReserved(allocKey string) bool {
+	sa.RLock()
+	defer sa.RUnlock()
+	return sa.reservations[allocKey] != nil
+}
+
+// getAllocationReservation returns the reservation object for the allocation
 // No locking must be called while holding the lock
-func (sa *Application) GetAskReservations(allocKey string) []string {
-	reservationKeys := make([]string, 0)
-	if allocKey == "" {
-		return reservationKeys
-	}
-	for key := range sa.reservations {
-		if strings.HasSuffix(key, allocKey) {
-			reservationKeys = append(reservationKeys, key)
-		}
-	}
-	return reservationKeys
+func (sa *Application) getAllocationReservation(allocKey string) *reservation {
+	return sa.reservations[allocKey]
```

Comment:
Do we need this method? It's just a map lookup. It's only used once in the prod code.

Reply:
removed